### DISCUSSION TITLE
[Bloc] Métodos sem estado do Geolocator

### DISCUSSION THREAD

**Discussion by brunosemfio** [Flutterando/forum#108] [Perguntas e Respostas] [answered]
Olá, pessoas.

O geolocator tem uns métodos bem "coisa de view":

```Geolocator.openAppSettings()``` e ```Geolocator.openLocationSettings()```

A lógica de verificar permissão e buscar a localização está num bloc. Mas esses métodos para abrir as configurações não têm estado, então não acho que faça sentido colocá-los no mesmo lugar.

Pensando uma arquitetura limpa, faz sentido chamar esses métodos direto na widget?

**Comment by eduardoflorence** [comment, marked as answer]
Olá @brunosemfio,
    Numa arquitetura limpa, o ideal é criar um Service com métodos que chamam esses dois métodos que citou.
    Depois você injeta esse service na sua view através do construtor. 
    Isso é importante para facilitar a construções de testes de unidade e também para desacoplar sua view de packages que, no futuro, você possa querer trocar.
    Normalmente você não injetaria um service na view, mas esses dois métodos que citou só tem função de redirecionamento para a parte de configurações de um celular, então não vejo problema, pois é como se fosse uma navegação.
